pkg/server/container_stop.go

```diff
@@ -138,33 +134,25 @@ func (c *criContainerdService) stopContainer(ctx context.Context, container cont
 
 // waitContainerStop polls container state until timeout exceeds or container is stopped.
 func (c *criContainerdService) waitContainerStop(ctx context.Context, id string, timeout time.Duration) error {
-	ticker := time.NewTicker(stopCheckPollInterval)
-	defer ticker.Stop()
 	timeoutTimer := time.NewTimer(timeout)
 	defer timeoutTimer.Stop()
-	for {
-		// Poll once before waiting for stopCheckPollInterval.
-		container, err := c.containerStore.Get(id)
-		if err != nil {
-			if err != store.ErrNotExist {
-				return fmt.Errorf("failed to get container %q: %v", id, err)
-			}
-			// Do not return error here because container was removed means
-			// it is already stopped.
-			logrus.Warnf("Container %q was removed during stopping", id)
-			return nil
-		}
-		// TODO(random-liu): Use channel with event handler instead of polling.
-		if container.Status.Get().State() == runtime.ContainerState_CONTAINER_EXITED {
-			return nil
-		}
-		select {
-		case <-ctx.Done():
-			return fmt.Errorf("wait container %q is cancelled", id)
-		case <-timeoutTimer.C:
-			return fmt.Errorf("wait container %q stop timeout", id)
-		case <-ticker.C:
-			continue
+	// Poll once before waiting for stopCheckPollInterval.
+	container, err := c.containerStore.Get(id)
```

Comment:
Can we pass `container containerstore.Container` to the `waitContainerStop` function to avoid this?

Reply:
Yean, it's better, will do.

Reply:
Done.